### PR TITLE
RavenDB-23057 migrated to CPM

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-23057

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [ ] Low 
- [x] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [ ] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -5,6 +5,13 @@
...
     <add key="project" value="libs" />
     <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
-    <add key="bdn-nightly" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />
   </packageSources>
+  <packageSourceMapping>
+    <packageSource key="project">
+      <package pattern="*" />
+    </packageSource>
+    <packageSource key="api.nuget.org">
+      <package pattern="*" />
```
Comment (gregolsky) [inline NuGet.Config:14]: If you have stars here, does it do anything? As I understand, this is to specify the package origin on a per dependency basis?

Comment (ppekrol) [reply NuGet.Config:14]: it is a name pattern a.k.a. 'what package can be downloaded from that source'

File: src/Raven.Client/Raven.Client.csproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Version></Version>
+    <Version>
+    </Version>
```
Comment (gregolsky) [inline src/Raven.Client/Raven.Client.csproj]: Are you sure splitting this into 2 lines won't break some regex value update here?

Comment (gregolsky) [reply src/Raven.Client/Raven.Client.csproj]:
```
$ grep '<Version>' -R  scripts/*
grep: scripts/assets/bin/nuget.exe: binary file matches
scripts/updateSourceWithBuildInfo.ps1:    $versionPattern = [regex]'(?sm)<Version>[A-Za-z0-9-\.\r\n\s]*</Version>'
scripts/updateSourceWithBuildInfo.ps1:    $result = $versionPattern.Replace($inputText, "<Version>$version</Version>")
```

:)

**Review by gregolsky — CHANGES_REQUESTED** [review]
please see my comments

**Comment by gregolsky** [comment]
cc @PFYasu @lukaszdobrzynski some takeaways for our projects here

File: src/Raven.Client/Raven.Client.csproj (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Client/Raven.Client.csproj]: Fixed

**Review by gregolsky — APPROVED** [review]

**Pull request merged**